### PR TITLE
Restrict deletion on group and relay group relations

### PR DESCRIPTION
**Pull request by vanstee**
Instead of restricting on delete for all foreign key relations, it made sense to only restrict relations that can be manually modified via the api. The main exception is the tree of bundle relations, so when a bundle is deleted we still cascade and delete all associated child records. To cleanly delete bundles, permissions are unlinked from any roles they had been granted to first. All exceptions raised are returned as errors on the id attribute since there's no other key that would really make sense to hang it off of.

Closes https://github.com/operable/cog/issues/794

### CODE REVIEW

File: lib/cog/models/relay_group_assignment.ex [hunk 1]
```diff
@@ -10,12 +10,10 @@ defmodule Cog.Models.RelayGroupAssignment do
   end
 
   @required_fields ~w(bundle_id group_id)
-  @optional_fields ~w()
 
   def changeset(model, params) do
     model
-    |> cast(params, @required_fields, @optional_fields)
-    |> foreign_key_constraint(:bundle_id, name: :relay_group_assignments_bundle_id_fkey)
+    |> cast(params, @required_fields)
```
Comment (vanstee) [inline lib/cog/models/relay_group_assignment.ex]: Removed this by mistake. Let me revert these changes.

File: test/cog/bootstrap_test.exs [hunk 2]
```diff
@@ -78,31 +74,34 @@ defmodule Cog.Bootstrap.Test do
...
     Cog.Bootstrap.bootstrap(@admin_user)
-    assert Cog.Models.Group |> Cog.Repo.one! |> Map.get(:name) == Cog.Util.Misc.admin_group
+    assert Group |> Repo.one! |> Map.get(:name) == Cog.Util.Misc.admin_group
     assert Cog.Bootstrap.is_bootstrapped?
   end
 
   test "the admin group cannot be deleted" do
     try do
-      Cog.Models.Group |> Cog.Repo.delete_all
+      admin_group = Repo.get_by!(Group, name: "cog-admin")
```
Comment (christophermaier) [inline test/cog/bootstrap_test.exs]: Should use `Cog.Util.Misc.admin_group` here for consistency.
